deployments/container/Dockerfile

```
@@ -116,13 +129,15 @@ RUN set -eux; \
 # - The extracted deb packages
 # - The extracted rpm packages
 # - The nvidia-ctk-installer binary
-FROM nvcr.io/nvidia/cuda:12.9.0-base-ubi9 AS artifacts
+FROM scratch AS artifacts
 
 COPY --from=rpmpackages /artifacts/rpm /artifacts/rpm
 COPY --from=debpackages /artifacts/deb /artifacts/deb
 COPY --from=build /artifacts/bin /artifacts/build
 
-FROM nvcr.io/nvidia/cuda:12.9.0-base-ubi9
+# The application stage contains the application used as a GPU Operator
+# operand.
+FROM nvcr.io/nvidia/distroless/go:v3.1.9-dev AS application
```

Note, the shell included in the `-dev` tags is located at `/busybox/sh`. I would recommend creating a symlink at `/bin/sh` so that in the operator we can use `#! /bin/sh` as the shebang for the entrypoint script. By using `/bin/sh` we remain backwards compatible with older toolkit images that are not built on distroless. We have tested this with other operands, e.g. https://github.com/NVIDIA/k8s-kata-manager/blob/f58e4dad0695043a545b17e3e159e24828816a62/deployments/container/Dockerfile#L50-L51
```suggestion
FROM nvcr.io/nvidia/distroless/go:v3.1.9-dev AS application

SHELL ["/busybox/sh", "-c"]
RUN ln -s /busybox/sh /bin/sh
```

---

Also, we may need to explicitly set `USER 0:0` in the Dockerfile as the default user in distroless is uid 1000. I assume the toolkit requires running as root (for restarting containerd).

---

Thanks for the shell tip. Will update.

I'm not sure on the user preference. Does the GPU Operator not set the user in general? Would using the current user (`USER 1000:1000`) not be more "compliant"?

---

I've updated the user to `0:0` below.

---

I have raised https://github.com/NVIDIA/gpu-operator/pull/1496 which updates our entrypoint scripts in the operator to use `sh` instead of `bash`.

---

I have updated the image to include a `/work/nvidia-toolkit` -> `/work/nvidia-ctk-installer` symlink. This should allow compatibility with the GPU Operator.